tests/codegen/autodiff/abi_handling.rs

```diff
@@ -0,0 +1,293 @@
+// CHECK: ; abi_handling::f3
+// CHECK-NEXT: ; Function Attrs: {{.*}}noinline{{.*}}
+// debug-NEXT: define internal float @_ZN12abi_handling2f317h9cd1fc602b0815a4E
+// debug-SAME: (ptr align 4 %x, ptr align 4 %y)
+// release-NEXT: define internal fastcc noundef float @_ZN12abi_handling2f317h9cd1fc602b0815a4E
+// release-SAME: (float %x.0.val)
+#[autodiff_forward(df3, Dual, Dual, Dual)]
+fn f3<'a>(x: &'a f32, y: &'a f32) -> f32 {
+    *x * *y
+}
+
+// CHECK: ; abi_handling::f4
+// CHECK-NEXT: ; Function Attrs: {{.*}}noinline{{.*}}
+// debug-NEXT: define internal float @_ZN12abi_handling2f417h2f4a9a7492d91e9fE
+// debug-SAME: (float %x.0, float %x.1)
+// release-NEXT: define internal fastcc noundef float @_ZN12abi_handling2f417h2f4a9a7492d91e9fE
+// release-SAME: (float noundef %x.0, float noundef %x.1)
+#[autodiff_forward(df4, Dual, Dual)]
+fn f4(x: (f32, f32)) -> f32 {
+    x.0 * x.1
+}
+
+// CHECK: ; abi_handling::f5
+// CHECK-NEXT: ; Function Attrs: {{.*}}noinline{{.*}}
+// debug-NEXT: define internal float @_ZN12abi_handling2f517hf8d4ac4d2c2a3976E
+// debug-SAME: (float %i.0, float %i.1)
+// release-NEXT: define internal fastcc noundef float @_ZN12abi_handling2f517hf8d4ac4d2c2a3976E
+// release-SAME: (float noundef %i.0, float noundef %i.1)
+#[autodiff_forward(df5, Dual, Dual)]
+fn f5(i: Input) -> f32 {
+    i.x + i.y
+}
+
+// CHECK: ; abi_handling::f6
+// CHECK-NEXT: ; Function Attrs: {{.*}}noinline{{.*}}
+// debug-NEXT: define internal float @_ZN12abi_handling2f617h5784b207bbb2483eE
+// debug-SAME: (float %i.0, float %i.1)
+// release-NEXT: define internal fastcc noundef float @_ZN12abi_handling2f617h5784b207bbb2483eE
+// release-SAME: (float noundef %i.0, float noundef %i.1)
+#[autodiff_forward(df6, Dual, Dual)]
+fn f6(i: NestedInput) -> f32 {
+    i.x + i.y.z * i.y.z
+}
+
+// df1
+// release: define internal fastcc { float, float }
+// release-SAME: @fwddiffe_ZN12abi_handling2f117h536ac8081c1e4101E
+// release-SAME: (float %x.0.val, float %x.4.val)
+// release-NEXT: start:
+// release-NEXT: %_0 = fadd float %x.0.val, %x.4.val
+// release-NEXT: %0 = insertvalue { float, float } undef, float %_0, 0
+// release-NEXT: %1 = insertvalue { float, float } %0, float 1.000000e+00, 1
+// release-NEXT: ret { float, float } %1
+// release-NEXT: }
+
+// debug: define internal { float, float } @fwddiffe_ZN12abi_handling2f117h536ac8081c1e4101E
+// debug-SAME: (ptr align 4 %x, ptr align 4 %"x'")
+// debug-NEXT: start:
+// debug-NEXT: %"'ipg" = getelementptr inbounds float, ptr %"x'", i64 0
+// debug-NEXT: %0 = getelementptr inbounds nuw float, ptr %x, i64 0
+// debug-NEXT: %"_2'ipl" = load float, ptr %"'ipg", align 4, !alias.scope !4, !noalias !7
+// debug-NEXT: %_2 = load float, ptr %0, align 4, !alias.scope !7, !noalias !4
+// debug-NEXT: %"'ipg2" = getelementptr inbounds float, ptr %"x'", i64 1
+// debug-NEXT: %1 = getelementptr inbounds nuw float, ptr %x, i64 1
+// debug-NEXT: %"_5'ipl" = load float, ptr %"'ipg2", align 4, !alias.scope !4, !noalias !7
```

Comment:
remove `, !alias.scope !4, !noalias !7` and similar scope and metadata annotations. align can stay.

They are fragile, numbers might change and we want to avoid test failures because of it.